WalletWasabi.Tests/UnitTests/AllFeeEstimateTests.cs

```diff
@@ -218,18 +267,37 @@ private static MockRpcClient CreateAndConfigureRpcClient(Func<int, FeeRate> esti
 					}),
 				OnGetPeersInfoAsync = async () =>
 					await Task.FromResult(hasPeersInfo ? new[] { new PeerInfo() } : Array.Empty<PeerInfo>()),
-
-				OnGetMempoolInfoAsync = async () =>
-					await Task.FromResult(new MemPoolInfo
-					{
-						MemPoolMinFee = memPoolMinFee // 1 s/b (default value)
-					}),
+				OnGetMempoolInfoAsync = async () => await Task.FromResult(ParseMempoolInfo(mempoolMinFee)),
 				OnEstimateSmartFeeAsync = async (target, _) =>
 					await Task.FromResult(new EstimateSmartFeeResponse
 					{
 						Blocks = target,
 						FeeRate = estimator(target)
 					})
 			};
+
+		private static MemPoolInfo ParseMempoolInfo(double mempoolMinFee)
+		{
+			var mempoolInfoWithHistogram = File.ReadAllText("./UnitTests/Data/MempoolInfoWithHistogram.json");
+			var jo = JObject.Parse(mempoolInfoWithHistogram);
+			var feeHistogram = (JObject)jo["fee_histogram"];
+
+			return new MemPoolInfo()
+			{
+				MemPoolMinFee = mempoolMinFee,
+				Histogram = feeHistogram.Properties()
+					.Where(p => p.Name != "total_fees")
+					.Select( p => new FeeRateGroup
+					{
+						Group = int.Parse(p.Name),
+						Sizes = p.Value.Value<ulong>("sizes"),
+						Count = p.Value.Value<uint>("count"),
+						Fees = Money.Satoshis(p.Value.Value<ulong>("fees")),
+						From = new FeeRate(p.Value.Value<decimal>("from_feerate")),
+						To = new FeeRate(Math.Min(100_000, p.Value.Value<decimal>("to_feerate")))
```

Comment:
I don't follow. So `p.Value.Value<decimal>("to_feerate")` gives you decimal value which is defined as follows:

> The binary representation of a Decimal value consists of a 1-bit sign, a 96-bit integer number, and a scaling factor used to divide the 96-bit integer and specify what portion of it is a decimal fraction. 

and 

> Decimal.MaxValue                79,228,162,514,264,337,593,543,950,335
> Decimal.MinValue               -79,228,162,514,264,337,593,543,950,335

so one can put decimal point here:

```
-79,228,162,514,264,337,593,5[separator]43,950,335
```

so you can express satoshis (after separator) and remains much more than 21m for Bitcoins (before separator). So even when you multiply with 1000 as in FeeRate, you are not out of range, are you?